Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1833,78 +1835,153 @@ inline void SetImage(CGImageRef image) {
 
 /**
  @Status Caveat
- @Notes Limited bitmap formats available. Decode, shouldInterpolate, intent parameters
- and some byte orders ignored.
- */
+ @Notes If the image colorspace is not a valid render target format (as per
+ https://msdn.microsoft.com/en-us/library/windows/desktop/dd756766(v=vs.85).aspx#supported_wic_formats), it's converted to 32bpp BGRA
+ format. Does not support conversion if data is already provided.
+*/
 CGContextRef CGBitmapContextCreate(void* data,
                                    size_t width,
                                    size_t height,
                                    size_t bitsPerComponent,
                                    size_t bytesPerRow,
                                    CGColorSpaceRef colorSpace,
                                    CGBitmapInfo bitmapInfo) {
-    UNIMPLEMENTED();
-    return nullptr;
+    return CGBitmapContextCreateWithData(data, width, height, bitsPerComponent, bytesPerRow, colorSpace, bitmapInfo, nullptr, nullptr);
+}
+
+/**
+ @Status Caveat
+ @Notes releaseCallback and releaseInfo is ignored. Also if the image colorspace is not a valid render target format (as per
+ https://msdn.microsoft.com/en-us/library/windows/desktop/dd756766(v=vs.85).aspx#supported_wic_formats), it's converted to 32bpp BGRA
+ format. Does not support conversion if data is already provided.
+*/
+CGContextRef CGBitmapContextCreateWithData(void* data,
+                                           size_t width,
+                                           size_t height,
+                                           size_t bitsPerComponent,
+                                           size_t bytesPerRow,
+                                           CGColorSpaceRef space,
+                                           uint32_t bitmapInfo,
+                                           CGBitmapContextReleaseDataCallback releaseCallback,
+                                           void* releaseInfo) {
+    RETURN_NULL_IF(!width);
+    RETURN_NULL_IF(!height);
+    RETURN_NULL_IF(!space);
+
+    size_t bitsPerPixel = ((bytesPerRow / width) << 3);
+    REFGUID pixelFormat = _CGImageGetWICPixelFormatFromImageProperties(bitsPerComponent, bitsPerPixel, space, bitmapInfo);
+
+    ComPtr<IWICBitmap> customBitmap;
+
+    if (_CGIsValidRenderTargetPixelFormat(pixelFormat)) {
+        // if data is null, enough memory is allocated via CGIWICBitmap
+        customBitmap = Make<CGIWICBitmap>(data, pixelFormat, height, width);
+    } else {
+        // TODO #<GITHUB-ID>: this will be an issue if we have change in stride, account for that.
+        // Also as per documentation, it's best to leave CGBitmapContext to manage the memory
+        if (data != nullptr) {
+            return nullptr;
+        }
```

Comment:
we have the functionality to convert, but we still need a direct access to the backed buffer (WIP)

---
In reply to: [87855777](https://github.com/Microsoft/WinObjC/pull/1371#discussion_r87855777) [](ancestors = 87855777)